From fea1a5175009682b84996e4f338c87eba6bbb9f5 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben.hutchings@codethink.co.uk>
Date: Tue, 23 Jun 2020 22:29:23 +0100
Subject: [PATCH 2/2] Add a udeb (Closes: #774797)

Now that we have two binary packages, debhelper will install to debian/tmp
and we need to define what gets copied from there to each binary package.

* For bcache-tools, copy everything (debian/tmp/*)
* For bcache-tools-udeb, copy only the udev rules and the binaries
---
 debian/bcache-tools-udeb.install | 2 ++
 debian/bcache-tools.install      | 1 +
 debian/changelog                 | 1 +
 debian/control                   | 7 +++++++
 4 files changed, 11 insertions(+)
 create mode 100644 debian/bcache-tools-udeb.install
 create mode 100644 debian/bcache-tools.install

diff --git a/debian/bcache-tools-udeb.install b/debian/bcache-tools-udeb.install
new file mode 100644
index 000000000000..fcde3b1ce05c
--- /dev/null
+++ b/debian/bcache-tools-udeb.install
@@ -0,0 +1,2 @@
+lib/udev/*
+usr/sbin/*
diff --git a/debian/bcache-tools.install b/debian/bcache-tools.install
new file mode 100644
index 000000000000..d2d93a94827a
--- /dev/null
+++ b/debian/bcache-tools.install
@@ -0,0 +1 @@
+debian/tmp/* /
diff --git a/debian/changelog b/debian/changelog
index 7af0fb2844f0..4a0ce6435441 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -2,6 +2,7 @@ bcache-tools (1.0.8-4.1) UNRELEASED; urgency=medium
 
   * Non-maintainer upload.
   * Create required directories from Makefile, not through dh_installdirs
+  * Add a udeb (Closes: #774797)
 
  -- Ben Hutchings <ben.hutchings@codethink.co.uk>  Tue, 23 Jun 2020 21:51:29 +0100
 
diff --git a/debian/control b/debian/control
index 634657cc87f2..0236b77ac508 100644
--- a/debian/control
+++ b/debian/control
@@ -22,3 +22,10 @@ Description: bcache userspace tools
  .
  This package includes udev rules, initramfs support, and the utilities to
  create a new bcache as well as inspect existing bcache partitions.
+
+Package: bcache-tools-udeb
+Section: debian-installer
+Architecture: linux-any
+Package-Type: udeb
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: bcache userspace tools, for debian-installer
